### ISSUE TITLE
`turbo watch` doesn't start tasks in order

### ISSUE THREAD

**Issue by kevva** [vercel/turbo#8899]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://codesandbox.io/p/devbox/d4gmxh?migrateFrom=fxlctj

### What package manager are you using / does the bug impact?

Yarn v2/v3/v4 (node_modules linker only)

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

2.0.11 (there is no canary after)

### Describe the Bug

When starting a task, let's say `dev`, in watch mode that depends on some packages with `build` tasks and `dev` is a persistent task, it will start immediately without waiting for its dependencies to build. This is quite annoying since I run a Next.js app in `dev` and it imports something in the instrumentation hook that depends on a package being built. But since the server starts immediately it will always throw because Next.js doesn't recover from failures in instrumentation.

### Expected Behavior

I would expect it to behave like running without watch mode enabled, i.e. that all tasks waits for their dependencies to be fulfilled before starting even though they're persistent.

### To Reproduce

In the reproduction example, run the following in the terminal:

```
yarn dlx turbo watch build --filter=app-a...
```

Notice how the `build` task for the `app-a` is started immediately even though it has dependencies that needs to be built.

**Comment by chris-olszewski** [comment]
Duplicate of #8673
